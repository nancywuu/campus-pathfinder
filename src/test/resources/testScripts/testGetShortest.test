CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D

AddEdge simpleGraph A B 10.0
AddEdge simpleGraph A C 13.0
AddEdge simpleGraph B C 2.0
AddEdge simpleGraph C D 3.0
AddEdge simpleGraph B D 7.0

FindPath simpleGraph B D

FindPath simpleGraph A D

FindPath simpleGraph A C