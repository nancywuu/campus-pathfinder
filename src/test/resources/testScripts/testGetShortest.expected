created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph

added edge 10.000 from A to B in simpleGraph
added edge 13.000 from A to C in simpleGraph
added edge 2.000 from B to C in simpleGraph
added edge 3.000 from C to D in simpleGraph
added edge 7.000 from B to D in simpleGraph

path from B to D:
B to C with weight 2.000
C to D with weight 3.000
total cost: 5.000

path from A to D:
A to B with weight 10.000
B to C with weight 2.000
C to D with weight 3.000
total cost: 15.000

path from A to C:
A to B with weight 10.000
B to C with weight 2.000
total cost: 12.000